core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/workflow/impl/SamplesReceivedWorkflow.java

```diff
@@ -0,0 +1,716 @@
+
+    private BoxProgressStep boxStep;
+
+    private int numAliquots;
+    private String error;
+
+    @Override
+    public WorkflowStepPrompt getPrompt() {
+      return PROMPT;
+    }
+
+    @Override
+    public ProgressStep getProgressStep() {
+      return boxStep;
+    }
+
+    @Override
+    public void cancelInput() {
+      boxStep = null;
+    }
+
+    @Override
+    public String getLogMessage() {
+      return String.format("Place Aliquots in Box '%s'", getAlias());
+    }
+
+    @Override
+    public void processInput(BoxProgressStep step) {
+      if (isBoxBigEnough(step)) {
+        boxStep = step;
+      } else {
+        error = String.format("Box '%s' does not have enough space for %d Aliquots (Box only has %d free spaces)",
+            step.getInput().getAlias(), numAliquots, step.getInput().getFreeCount());
+      }
+    }
+
+    public String getAlias() {
+      return boxStep.getInput().getAlias();
+    }
+
+    public boolean isBoxBigEnough(BoxProgressStep step) {
+      return step.getInput().getFreeCount() >= numAliquots;
+    }
+
+    public Box getBox() {
+      return boxStep.getInput();
+    }
+
+    @Override
+    public String getError() {
+      return error;
+    }
+
+    public void setNumAliquots(int numAliquots) {
+      this.numAliquots = numAliquots;
+    }
+  }
+
+  private static class AliquotStep implements WorkflowStep {
+
+    private static final WorkflowStepPrompt PROMPT = new WorkflowStepPrompt(Sets.newHashSet(InputType.POSITIVE_INTEGER),
+        "Enter the number of Aliquots");
+
+    private PositiveIntegerProgressStep aliquotStep;
+
+    @Override
+    public WorkflowStepPrompt getPrompt() {
+      return PROMPT;
+    }
+
+    @Override
+    public ProgressStep getProgressStep() {
+      return aliquotStep;
+    }
+
+    @Override
+    public void cancelInput() {
+      aliquotStep = null;
+    }
+
+    @Override
+    public String getLogMessage() {
+      return String.format("Create %d Aliquot(s)", getAliquotQuantity());
+    }
+
+    @Override
+    public void processInput(PositiveIntegerProgressStep step) {
+      aliquotStep = step;
+    }
+
+    public int getAliquotQuantity() {
+      return aliquotStep.getInput();
+    }
+
+  }
+  
+  private static class StockBoxPositionStep implements WorkflowStep {
+
+    private BoxPositionProgressStep boxPositionStep;
+
+    private final Box box;
+    private String firstFreePosition;
+
+    private String error;
+
+    StockBoxPositionStep(Box box) {
+      this.box = box;
+    }
+
+    @Override
+    public WorkflowStepPrompt getPrompt() {
+      firstFreePosition = getFirstFreePosition();
+      return new WorkflowStepPrompt(Sets.newHashSet(InputType.BOX_POSITION, InputType.SKIP),
+          String.format("Enter a Box Position to place the Stock.\nSkipping will default to position %s", firstFreePosition));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') %n should be used in place of \n to produce the platform-specific line separator. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3457)